common/common.go

```diff
@@ -158,6 +161,15 @@ const (
 	CacheVersion = "1.0.0"
 )
 
+// GetGnuPGHomePath retrieves the path to use for GnuPG home directory, which is either taken from GNUPGHOME environment or a default value
```

Comment:
Might need to make this more configurable; e.g. for compatibility with KSOPS
ideas:
  - Check some variable `GNUPGHOME_ARGOCD_TRUST` first?
  - Have a configuration variable `ARGOCD_GNUPGHOME_ENV` which contains the actual env var to check (and default it to `GNUGPGHOME`)

Reply:
Since we will only manage public GnuPG keys and perform signature verification, I think stuff like KSOPS will be way out of scope for this change.

Reply:
The issue I'm thinking of is that KSOPS also pays attention to the `GNUPGHOME` env var (see https://github.com/viaduct-ai/kustomize-sops/issues/24 )

But yeah, if we aren't using the trust db for the verification then things should be fine.

Reply:
I was thinking about that for a while, and came to the conclusion that your concerns are valid. I think the code should not use `GNUPGHOME`, as it could be set from someone else for whatever tooling they are incorporating. 

I think `ARGOCD_GNUPGHOME` should be sufficient, because usually, this var should not be set and will be used only during testing (i.e. when running outside a k8s cluster, e.g. in CI) and `/app/config/gpg` is not available.

Thanks for bringing up that concern, much appreciated!

Reply:
The code now uses `ARGOCD_GNUPGHOME` and ignores `GNUPGHOME` completely as to not interfere with other custom `gpg` tooling by accident in the repo server.